src/components/ProFullCourt/index.tsx

```diff
@@ -0,0 +1,78 @@
+import { Stage, Layer, Group, Text } from "react-konva";
+import { Flex } from "@chakra-ui/react";
+import { ReactReduxContext, Provider } from "react-redux";
+import ThreePointArea from "../BasketballCourt/ThreePointArea";
+import KeyArea from "../BasketballCourt/KeyArea";
+import CourtArea from "../BasketballCourt/CourtArea";
+import CircleArea from "../BasketballCourt/CircleArea";
+import TopKeyArea from "../BasketballCourt/TopKeyArea";
+import Border from "../BasketballCourt/Border";
+import BorderDimensionLine from "../BasketballCourt/BorderDimensionLine";
+import ArrowLine from "../BasketballCourt/Arrow";
+import { useStoreSelector } from "@/store/hooks";
+
+const ProFullCourt = () => {
+  const { initPointX, courtAreaXLength } = useStoreSelector((state) => state.courtSize);
+  const courtRatio = 0.25;
+  return (
+    <Flex
+      position="fixed"
+      top="122px"
+      left="98px"
+      width="calc(100% - 98px)"
+      height="calc(100% - 162px)"
+      minWidth={850}
+      minHeight={600}
+      justifyContent="center"
+      alignItems="center"
+      margin="auto"
+    >
+      <ReactReduxContext.Consumer>
+        {({ store }) => (
+          <Stage
+            id="basketball-court"
+            width={850}
+            height={600}
+            visible={true}
+            style={{ backgroundColor: "white" }}
+          >
+            <Provider store={store}>
+              <Layer>
+                <Text
+                  width={850}
+                  height={100}
+                  text={"Pro Full Court: 510 ㎡ (17 * 30)"}
+                  fontSize={150 * courtRatio}
+                  align="center"
+                  fill="black"
+                  x={0}
+                  y={40}
+                />
+                <Border courtRatio={courtRatio} color={"#195955"} />
+                <ArrowLine courtRatio={courtRatio} arrowXEndLength={courtAreaXLength} />
+                <Group>
+                  <BorderDimensionLine courtRatio={courtRatio} />
+                  <CourtArea courtRatio={courtRatio} color={"#B61313"} />
+                  <ThreePointArea courtRatio={courtRatio} color={"#72818B"} />
+                  <KeyArea courtRatio={courtRatio} color={"#2C4E8A"} />
+                  <CircleArea courtRatio={courtRatio} color={"#606F14"} />
+                  <TopKeyArea courtRatio={courtRatio} color={"#B61313"} />
+                </Group>
+                <Group scaleX={-1} x={initPointX * 2 + courtAreaXLength * 2 * courtRatio}>

```

Comment:
In this part, if I'm not wrong, you pass the formula result to the whole group and combine using scaleX to flip the half of the court horizontally?

Do you think is there any easy-to-understand way of doing it?
It is just a though, not a change request.

Reply:
Sure. now I am still looking for a better way to simplify the code. A fix branch may be added once an easy way has been found to achieve the goal. Cheers~~~